Azure Pipeline failing at react-native-windows/Directory.Build.targets(78,5) with 'Syntax of Time-Parameter is wrong', Local build works

Hello there,

We're close to releasing our RN-Windows App to the wild and are setting up the ci/cd pipeline with azure.
Unfortunately we were not able to use the MS-Hosted agents due to 10gb size limit (Idk why, my local build folder is only 2.6gb big) so I setup a Self-Hosted Agent on my working machine.

Whenever I build via the pipeline it succeeds almost to the end until the step `CleanUpPrecompHeaders ` and aborts there with:
```
_GenerateProjectPriFileCore:
 C:\Program Files (x86)\Windows Kits\10\bin\10.0.18362.0\x64\MakePri.exe New -ProjectRoot C:\agent\_work\1\s\node_modules\react-native-windows\Microsoft.ReactNative\ -ConfigXml C:\agent\_work\1\s\node_modules\react-native-windows\\build\x64\Release\Microsoft.ReactNative\priconfig.xml -OutputFile C:\agent\_work\1\s\node_modules\react-native-windows\target\x64\Release\Microsoft.ReactNative\Microsoft.ReactNative.pri -IndexName Microsoft.ReactNative -Verbose -Overwrite  
CleanUpPrecompHeaders:
  The file "C:\agent\_work\1\s\node_modules\react-native-windows\build\x64\Release\Microsoft.ReactNative\pch.pch" is being deleted.
##[error]node_modules\react-native-windows\Directory.Build.targets(78,5): Error MSB3376: The syntax of the Time parameter is incorrect. Die Zeichenfolge wurde nicht als gültiges DateTime erkannt.
```

That's a generated datetime and I'm pretty clueless on what's going wrong here, the file states though that: `As a cppwinrt project, .pch is very big, and delete it in pipeline after build
    Instead of outright deleting the PCHs, we want to trick the "project freshness detector" by
    emitting empty files that look suspiciously like the PCHs it's expecting.
    It's of utmost importance that their timestamps match up`

So it seems they do not match up, but why? A Local build, not via the azure pipeline agent works flawlessly.

--- a/change/react-native-windows-51bbc131-2b1c-465c-b2a8-b06ce1e6b002.json
+++ b/change/react-native-windows-51bbc131-2b1c-465c-b2a8-b06ce1e6b002.json
@@ -1,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "Native build logic shouldn't delete PCH's on all CI machines",
+  "packageName": "react-native-windows",
+  "email": "ngerlem@microsoft.com",
+  "dependentChangeType": "patch"
+}

--- a/vnext/Directory.Build.targets
+++ b/vnext/Directory.Build.targets
@@ -56,27 +56,4 @@
     <Message Text="=> MSBuildExtensionsPath64 [$(MSBuildExtensionsPath64)]" />
     <Message Text="=> VCTargetsPath           [$(VCTargetsPath)]" />
   </Target>
-
-  <Target Name="_ComputePrecompToCleanUp">
-    <ItemGroup>
-      <PCHFileToClean Include="$(IntDir)\**\*.pch" />
-      <_PCHFileToCleanWithTimestamp Include="@(PCHFileToClean)" Condition="'%(Identity)' != ''">
-        <LastWriteTime>$([System.IO.File]::GetLastWriteTime('%(Identity)'))</LastWriteTime>
-      </_PCHFileToCleanWithTimestamp>
-    </ItemGroup>
-  </Target>
-  <!--
-    As a C++/WinRT project, .pch is very big, and delete it in pipeline after build
-    Instead of outright deleting the PCHs, we want to trick the "project freshness detector" by
-    emitting empty files that look suspiciously like the PCHs it's expecting.
-    It's of utmost importance that their timestamps match up. -->
-  <Target Name="CleanUpPrecompHeaders"
-          DependsOnTargets="_ComputePrecompToCleanUp"
-          AfterTargets="AfterBuild"
-          Condition="'$(AGENT_ID)' != ''">
-    <!-- We just need to keep *ReactNativeWindows*'s PCHs because they get rebuilt more often. -->
-    <Delete Files="@(_PCHFileToCleanWithTimestamp)"/>
-    <Touch Files="@(_PCHFileToCleanWithTimestamp)" Time="%(LastWriteTime)" AlwaysCreate="true" />
-    <Message Text="PCH and Precomp object @(_PCHFileToCleanWithTimestamp) has been deleted for $(ProjectName)." />
-  </Target>
 </Project>

